### ISSUE TITLE
Support traits with generics

### ISSUE THREAD

**Issue by dtolnay** [kardeiz/objekt-clonable#1]
https://docs.rs/objekt/0.1.2/objekt/macro.clone_trait_object.html shows how clone_trait_object supports traits with generic parameters or where-clauses.

```rust
use std::io::Read;

trait Difficult<R>: objekt::Clone where R: Read {
    /* ... */
}

objekt::clone_trait_object!(<R> Difficult<R> where R: Read);
```

There should be a way to do this with `#[clonable]`. The current error is:

```rust
use objekt_clonable::clonable;
use std::io::Read;

#[clonable]
trait Difficult<R>: Clone where R: Read {
    /* ... */
}
```

```console
error[E0107]: wrong number of type arguments: expected 1, found 0
```

**Comment by kardeiz** [comment]
Sorry, completely missed that in the `objekt` docs. Fixed in [`0.2.2`](https://crates.io/crates/objekt-clonable/0.2.2).

**Comment by dtolnay** [comment]
Nice, thanks for the quick fix.
